File: app/repository/ScoreRepository.java
package repository;

import models.Score;
import play.db.ebean.EbeanConfig;

import javax.inject.Inject;


/**
 * A repository that executes database operations in a different
 * execution context.
 */
public class ScoreRepository extends BaseRepository<Score> {

    @Inject
    public ScoreRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext executionContext) {
        super(ebeanConfig, executionContext);
    }

    @Override
    protected Class<Score> getModelClass() {
        return Score.class;
    }
}

File: app/repository/ScoreRepository.scala
package repository

import javax.inject._
import models.Score
import play.db.ebean.EbeanConfig

import scala.concurrent.Future

class ScoreRepository @Inject()(val ebeanConfig: EbeanConfig, val executionContext: DatabaseExecutionContext) extends BaseRepository[Score] {
  val modelClass: Class[Score] = classOf[Score]
}
